get-started-with-rubyonrails.md

```diff
@@ -0,0 +1,336 @@
+#SAVE STATE
+```
+
+Edit the database config file and add your MySQL username and MySQL database name and MySQL host URL
+
+```
+nano /opt/apps/myapp/config/database.yml
+```
+Create a database schema. If you have already created a database schema you can skip this.
+
+```
+rake db:create
+```
+
+Run all migration scripts to create the tables in the DB. This step needs to run everytime you makes change to table definitions
+
+```
+rake db:migrate
+```
+
+Start rails server as a background process, listen on port 8080 and bind all interfaces and send log output to startup.log
+
+```
+rails s -p 8080 -b 0.0.0.0 >> ./log/startup.log &
+```
+
+## Accessing the application
+If you configured your VM manually you should now be able to access your new rails application on:
+
+```bash
+http://my-vm-public.ip:8080
+```
+
+If you provisioned your environment automatically using the "Deploy Button" you can access the application through the load balancer: 
+
+```
+<http://my-load-balancer-ip>
+```
+
+Check Oracle Resource Manager output for the ip or check the loadbalancer page.
+
+The loadbalancer is configured to listen on port 80 and forwards all traffic to the VM where the Ruby on Rails application is running.
+
+The load balancer will forward traffic to all the VMs if you provisioned more than one VM.  
+
+# Summary 
+A new Rails App is created in the directory: ```$/opt/apps/myapp```
+
+A database schema is created on the MySQL server.
+
+Database schema name: ```myapp```
+
+The app is configured to connect to the database. Configuration can be found in: ```/opt/apps/app/config/database.yml```
+
+# How to deploy your existing application
+
+Before deleting the app we will take a backup of the database.yml file. We need it later.
+
+```bash
+cp /opt/apps/myapp/config/database.yml  /opt/apps/bck_database.yml
+```
+
+1. Delete the existing vanilla Ruby on Rails app created in 
+
+```
+/opt/apps/myapp
+```
+
+```bash
+rm -rf /opt/apps/myapp
+```
+
+2. Copy your application to the server using scp
+
+
+```bash
+scp myapp.zip ubuntu@server-ip:/opt/apps/
+```
+
+3. Unpack the application 
+
+```bash
+unzip myapp.zip ./myapp/
+```
+
+4. Copy the generated database config file
+
+```
+cp /opt/apps/bck_database.yml ./myapp/config/database.yml
+```
+
+5. Or edit the existing database config file
+
+```Configure the database.yml:
+username: 
+password: 
+url: 
+```
+
+6. Migrate your data to the MySQL Database 
+
+7. Install dependencies 
+
+```bash
+bundle install
+```
+8. Run Rails migrations
+
+```
+rake db:migrate 
+```
+Start the server: 
+
+```bash
+Rails s -b 0.0.0.0 -p 8080
+```
+
+# How to change Ruby version
+The default ruby version installed is 3.0.1. If you need a different ruby version simply run: 
+
+```bash
+rbenv install 3.0.2
+```
+
+You can see available Ruby version by running: 
+- Rbenv install list 
+
+In the part we take a look at the following topics, stay tuned:
+
+## How to build a new application 
+- Scaffold 
```

Comment:
Formality: newline before list item is not strictly required after headline in GFM, but good practice for compatibility with all processors.